src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Security/AuthorizationHandlers/DeleteSpecificMethodologyAuthorizationHandlerTests.cs

```diff
@@ -0,0 +1,326 @@
+            {
+                Id = Guid.NewGuid(),
+                Status = Approved,
+                MethodologyParent = new MethodologyParent()
+            };
+        
+            [Fact]
+            public async Task UserWithCorrectClaimCanDeleteNonPublicDraftMethodology()
+            {
+                await ForEachSecurityClaimAsync(async claim =>
+                {
+                    await using var context = InMemoryApplicationDbContext(Guid.NewGuid().ToString());
+                    context.Attach(MethodologyNoPublications);
+
+                    var (handler, methodologyRepository, publicationRoleRepository) =
+                        CreateHandlerAndDependencies(context);
+
+                    methodologyRepository
+                        .Setup(s => s.IsPubliclyAccessible(MethodologyNoPublications.Id))
+                        .ReturnsAsync(false);
+
+                    var user = CreateClaimsPrincipal(UserId, claim);
+                    var authContext = CreateAuthContext(user, MethodologyNoPublications);
+
+                    await handler.HandleAsync(authContext);
+                    VerifyAllMocks(methodologyRepository, publicationRoleRepository);
+
+                    // Verify that the presence of the "DeleteAllMethodologies" Claim and no other will pass the handler test
+                    Assert.Equal(claim == DeleteAllMethodologies, authContext.HasSucceeded);
+                });
+            }
+            
+            [Fact]
+            public async Task UserWithCorrectClaimCannotDeletePubliclyAccessibleMethodology()
+            {
+                await ForEachSecurityClaimAsync(async claim =>
+                {
+                    await using var context = InMemoryApplicationDbContext(Guid.NewGuid().ToString());
+                    context.Attach(MethodologyNoPublications);
+
+                    var (handler, methodologyRepository, publicationRoleRepository) =
+                        CreateHandlerAndDependencies(context);
+
+                    methodologyRepository
+                        .Setup(s => s.IsPubliclyAccessible(MethodologyNoPublications.Id))
+                        .ReturnsAsync(true);
+
+                    var user = CreateClaimsPrincipal(UserId, claim);
+                    var authContext = CreateAuthContext(user, MethodologyNoPublications);
+
+                    await handler.HandleAsync(authContext);
+                    VerifyAllMocks(methodologyRepository, publicationRoleRepository);
+
+                    // Verify that the presence of the "DeleteAllMethodologies" Claim still doesn't allow the user to
+                    // do this with a Methodology in this state.
+                    Assert.False(authContext.HasSucceeded);
+                });
+            }
+            
+            [Fact]
+            public async Task UserWithCorrectClaimCannotDeleteApprovedMethodology()
```

Comment:
Not sure I follow what specifically about it being not approved is making this handler fail as I didn't spot anything about approved in the handler code. Isn't it `IsPubliclyAccessible` returning false which is making the handler fail here?

Reply:
If the Methodology is an Amendment or is the first Draft version, it can be deleted.  In this case, this is the first version but it is *not Draft*, and so it's not available for deletion unless it's unapproved.

Reply:
Thanks I understand now